src/Sidekick.Business/Apis/Poe/Models/Socket.cs

```diff
@@ -0,0 +1,41 @@
+using System;
+using System.Text.Json.Serialization;
+
+namespace Sidekick.Business.Apis.Poe.Models
+{
+    public class Socket
+    {
+        public int Group { get; set; }
+
+        [JsonPropertyName("sColour")]
+        public string __Color { get; set; }
+
+        [JsonIgnore]
+        public SocketColor Color
+        {
+            get
+            {
+                return __Color switch
+                {
+                    "B" => SocketColor.Blue,
+                    "G" => SocketColor.Green,
+                    "R" => SocketColor.Red,
+                    "W" => SocketColor.White,
+                    "A" => SocketColor.Abyss,
+                    _ => throw new Exception("Invalid socket"),
+                };
+            }
+            set
+            {
+                switch (value)
+                {
+                    case SocketColor.Blue: __Color = "B"; break;
+                    case SocketColor.Green: __Color = "G"; break;
+                    case SocketColor.Red: __Color = "R"; break;
+                    case SocketColor.White: __Color = "W"; break;
+                    case SocketColor.Abyss: __Color = "A"; break;
+                }
```

Comment:
The switch expression is for quickly returning data. Doing assignment calls is kind of tricky and readability is not really improved. The underscore is required because the switch expression expects a return value.
``` cs
                _ = (value switch
                {
                    SocketColor.Blue => __Color = "B",
                    SocketColor.Green => __Color = "G",
                    SocketColor.Red => __Color = "R",
                    SocketColor.White => __Color = "W",
                    SocketColor.Abyss => __Color = "A",
                    _ => throw new Exception("Invalid socket"),
                });
```

Reply:
__Color = value switch
	{
	    SocketColor.Blue => "B",
	    SocketColor.Green => "G",
	    SocketColor.Red => "R",
	    SocketColor.White => "W",
	    SocketColor.Abyss => "A",
	    _ => throw new Exception("Invalid socket"),
	};

This should work just fine, right?

Reply:
Oh yeah, just woke up...